library/src/main/java/com/novoda/storagepathfinder/DeviceStoragePath.java

```diff
@@ -0,0 +1,68 @@
+package com.novoda.storagepathfinder;
+
+import java.io.File;
+
+/**
+ * Wrapper around a string for a Storage path with a type.
+ */
+class DeviceStoragePath implements StoragePath {
+
+    public static StoragePath create(String path, DeviceStoragePath.Type type) {
+        return new DeviceStoragePath(path, type);
+    }
+
+    private final String path;
+    private final Type type;
+
+    private DeviceStoragePath(String path, Type type) {
+        this.path = path;
+        this.type = type;
+    }
+
+    @Override
+    public Type getType() {
+        return type;
+    }
+
+    @Override
+    public String asString() {
```

Comment:
`asString` implies you are getting the String of both the type and the path, this isn't the case. Opt for being more specific in this case.